src/autopas/containers/verletClusterLists/VerletClusterLists.h

```diff
@@ -383,57 +404,135 @@ class VerletClusterLists : public ParticleContainerInterface<Particle>, public i
     return TraversalSelectorInfo(towerDimensions, this->getInteractionLength(), towerSize, _clusterSize);
   }
 
+  /**
+   * Helper function for begin() and getRegionIterator() to check if the passed additionalVectors are empty
+   *
+   * @tparam modifiable
+   * @tparam regionIter
+   * @param additionalVectors
+   * @return true iff all passed vectors are empty
+   * @return false iff at least one buffer from the passed vectors is not empty
+   */
+  template <bool modifiable, bool regionIter>
+  bool additionalVectorsEmpty(
+      typename ContainerIterator<Particle, modifiable, regionIter>::ParticleVecType *additionalVectors) const {
+    bool addVectorsEmpty = true;
+    if (additionalVectors) {
+      for (auto &buffer : *additionalVectors) {
+        if (not buffer->empty()) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Helper function for begin() and getRegionIterator() that creates a ContainerIterator from all particles in
+   * _particlesToAdd
+   *
+   * @tparam regionIter
+   * @param additionalVectorsToPass
+   */
+  template <bool regionIter>
+  void createIteratorFromParticlesToAdd(
```

Comment:
I feel this name is somewhat misleading. The function doesn't create any iterator, it appends the vectors.

Reply:
done in [2cbb6b5](https://github.com/AutoPas/AutoPas/pull/753/commits/2cbb6b548bdbdf5c3fb851bb25a0170a172b0464)